contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,384 @@
+        address[] memory allStrategies = IVault(vaultAddress)
+            .getAllStrategies();
+        for (uint256 i = 0; i < allStrategies.length; i++) {
+            _harvest(allStrategies[i]);
+        }
+    }
+
+    /**
+     * @dev Collect reward tokens for a specific strategy. Called from the vault.
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function harvest(address _strategyAddr) external onlyGovernor nonReentrant {
+        // TODO: add protection so that harvestAndSwap isn't called twice too closely together
+        _harvest(_strategyAddr);
+    }
+
+    /**
+     * @dev Collect reward tokens for a specific strategy and swap for supported
+     *      stablecoin via Uniswap. Called from the vault.
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function harvestAndSwap(address _strategyAddr) external nonReentrant {
+        IStrategy strategy = IStrategy(_strategyAddr);
+        _harvest(address(strategy));
+        address[] memory rewardTokens = strategy.getRewardTokenAddresses();
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            _swap(rewardTokens[i]);
+        }
+        if (msg.sender != vaultAddress) {
+            IVault(vaultAddress).rebase();
+        }
+    }
+
+    /**
+     * @dev Collect reward tokens from a single strategy and swap them for a
+     *      supported stablecoin via Uniswap
+     * @param _strategyAddr Address of the strategy to collect rewards from.
+     */
+    function _harvest(address _strategyAddr) internal {
+        IStrategy strategy = IStrategy(_strategyAddr);
+        address[] memory rewardTokenAddresses = strategy
+            .getRewardTokenAddresses();
+
+        if (rewardTokenAddresses.length > 0) {
+            strategy.collectRewardTokens();
+        }
+    }
+
+    /**
+     * @dev Swap all supported swap tokens for stablecoins via Uniswap.
+     */
+    function _swap() internal {
+        address[] memory allStrategies = IVault(vaultAddress)
+            .getAllStrategies();
+
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            _swap(swapTokens[i]);
+        }
+    }
+
+    /**
+     * @dev Swap a record token for stablecoins for Uniswap. The token must have
+     *       a registered price feed with the price provider.
+     * @param _swapToken Address of the token to swap.
+     */
+    function _swap(address _swapToken)
+        internal
+        returns (uint256[] memory swapResult)
+    {
+        RewardTokenConfig memory tokenConfig = rewardTokenConfigs[_swapToken];
+        require(
+            tokenConfig.uniswapV2CompatibleAddr != address(0),
+            "Swap token is missing token configuration."
+        );
+
+        address priceProvider = IVault(vaultAddress).priceProvider();
+
+        IERC20 swapToken = IERC20(_swapToken);
+        uint256 balance = swapToken.balanceOf(address(this));
+        if (balance > 0) {
```

Comment:
Rather than indenting this entire section, let's just fast return here if balance is zero instead. Will make things clearer.  In case you can't tell I'm a big fan of fast returns over nested ifs. :D

Slither will yell, but it's in the wrong.

Reply:
Yup thanks for pointing it out, I prefer early exits as well.